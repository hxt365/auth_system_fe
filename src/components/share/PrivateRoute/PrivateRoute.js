import React, { useContext } from 'react';
import { Redirect, Route } from 'react-router-dom';
import { AppContext } from 'components/AppLayout';
import { LOGIN, HOME } from 'constants/route';

// eslint-disable-next-line react/prop-types
function PrivateRoute({ component: Component, authRoute, ...rest }) {
  const state = useContext(AppContext);

  return authRoute ? (
    <Route
      {...rest}
      render={props =>
        state.user.is_authenticated ? <Component {...props} /> : <Redirect to={LOGIN} />
      }
    />
  ) : (
    <Route
      {...rest}
      render={props =>
        !state.user.is_authenticated ? <Component {...props} /> : <Redirect to={HOME} />
      }
    />
  );
}

export default PrivateRoute;
