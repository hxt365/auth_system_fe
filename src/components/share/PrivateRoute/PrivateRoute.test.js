import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import PrivateRoute from './PrivateRoute';

describe('test PrivateRoute', () => {
  it('test snapshot', () => {
    const wrapper = shallow(<PrivateRoute />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
