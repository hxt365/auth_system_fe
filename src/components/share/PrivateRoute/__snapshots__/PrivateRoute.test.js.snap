// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PrivateRoute test snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
